src/components/Integrations/IntegrationDetails.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import {
+  DescriptionList,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+} from '@patternfly/react-core';
+import {
+  IntegrationCamel,
+  IntegrationIcon,
+  IntegrationPagerduty,
+  IntegrationType,
+} from '../../types/Integration';
+import { IntegrationStatus, StatusUnknown } from './Table/IntegrationStatus';
+import { IntegrationRow } from './Table';
+import Config, { defaultIconList } from '../../config/Config';
+import messages from '../../properties/DefinedMessages';
+import { useIntl } from 'react-intl';
+
+export const getIntegrationIcon = (type: string): React.ReactElement | null => {
+  const allIcons = Object.assign(
+    defaultIconList['Communications'],
+    defaultIconList['Reporting']
+  );
```

Comment:
No need to use Object.assign
```suggestion
  const allIcons = {
    ...defaultIconList['Communications'],
    ...defaultIconList['Reporting']
  };
```